go/vt/vtgate/engine/primitive.go

```diff
@@ -120,10 +120,17 @@ func (p *Plan) Size() int {
 	return 1
 }
 
+// KeyspaceTableName
+type KeyspaceTableName struct {
+	Keyspace string
+	Table    string
+}
+
 // Primitive is the interface that needs to be satisfied by
 // all primitives of a plan.
 type Primitive interface {
 	RouteType() string
+	KeyspaceTableNames() []*KeyspaceTableName
```

Comment:
Building on the other comment -- I actually think it would be clearer and simpler if this was split into a separate functions `KeyspaceName()` and `TableName()`, each of which returns a simple string.